src/nfa/operations.cc

```diff
@@ -505,11 +505,7 @@ bool mata::nfa::Nfa::is_lang_empty(Run* cex) const {
     //TOOD: hot fix for performance reasons for TACAS.
     // Perhaps make the get_useful_states return a witness on demand somehow.
     if (!cex) {
-        BoolVector useful_states = get_useful_states(true);
-        for (State is_state_useful: useful_states)
-            if (is_state_useful)
-                return false;
-        return true;
+        return is_lang_empty_scc();
```

Comment:
If `is_lang_empty` is going to select "the best" algorithm based on arguments, maybe we should take out the rest and make it its own function too? Like `is_lang_empty_basic` or something like that?

Reply:
I don't know. There were some changes in `is_lang_empty` before TACAS deadline. Not really sure why it happened, so I just keep my hands off this method.